lightning/src/ln/channelmanager.rs

```diff
@@ -2188,17 +2202,23 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 					self.fail_htlc_backwards_internal(channel_state,
 						htlc_src, &payment_hash, HTLCFailReason::Reason { failure_code, data: onion_failure_data});
 				},
-				HTLCSource::OutboundRoute { .. } => {
-					self.pending_events.lock().unwrap().push(
-						events::Event::PaymentFailed {
-							payment_hash,
-							rejected_by_dest: false,
+				HTLCSource::OutboundRoute { session_priv, .. } => {
+					if {
+						let mut session_priv_bytes = [0; 32];
+						session_priv_bytes.copy_from_slice(&session_priv[..]);
+						self.outbound_pending_payments.lock().unwrap().remove(&session_priv_bytes)
+					} {
```

Comment:
Fwiw, I commented these lines out and all tests passed

Reply:
Hmm, indeed, I don't know that its possible to have a duplicate fail issue for an HTLC that was never forwarded as it was in the holding cell. Could add a test which just ensures we dont "leak" any outbound payments, but that's all this does.